templates/org/settings/applications_edit.tmpl

```diff
@@ -0,0 +1,67 @@
+{{template "base/head" .}}
+<div class="page-content organization settings options">
+	{{template "org/header" .}}
+	<div class="ui container">
+		<div class="ui grid">
+			{{template "org/settings/navbar" .}}
+			<div class="twelve wide column content">
+				{{template "base/alert" .}}
+				<h4 class="ui top attached header">
+					{{.locale.Tr "settings.edit_oauth2_application"}}
+				</h4>
+				<div class="ui attached segment form ignore-dirty">
+					{{.CsrfTokenHtml}}
+					<div class="field">
+						<label for="client-id">{{.locale.Tr "settings.oauth2_client_id"}}</label>
+						<input id="client-id" readonly value="{{.App.ClientID}}">
+					</div>
+					{{if .ClientSecret}}
+						<div class="field">
+							<label for="client-secret">{{.locale.Tr "settings.oauth2_client_secret"}}</label>
+							<input id="client-secret" type="text" readonly value="{{.ClientSecret}}">
+						</div>
+					{{else}}
+						<div class="field">
+							<label for="client-secret">{{.locale.Tr "settings.oauth2_client_secret"}}</label>
+							<input id="client-secret" type="password" readonly value="averysecuresecret">
+						</div>
+					{{end}}
+					<div class="item">
+						{{.locale.Tr "settings.oauth2_regenerate_secret_hint"}}
+						<form class="ui form ignore-dirty" action="{{AppSubUrl}}/org/{{.Org.Name}}/settings/applications/{{.App.ID}}/regenerate_secret" method="post">
+							{{.CsrfTokenHtml}}
+							<a href="#" onclick="event.target.parentNode.submit()">{{.locale.Tr "settings.oauth2_regenerate_secret"}}</a>
+						</form>
+					</div>
+				</div>
+				<div class="ui attached bottom segment">
+					<form class="ui form ignore-dirty" action="{{AppSubUrl}}/org/{{.Org.Name}}/settings/applications/{{.App.ID}}" method="post">
+						{{.CsrfTokenHtml}}
+						<div class="field {{if .Err_AppName}}error{{end}}">
+							<label for="application-name">{{.locale.Tr "settings.oauth2_application_name"}}</label>
+							<input id="application-name" value="{{.App.Name}}" name="application_name" required>
+						</div>
+						<div class="field {{if .Err_RedirectURI}}error{{end}}">
+							<label for="redirect-uri">{{.locale.Tr "settings.oauth2_redirect_uri"}}</label>
+							<input type="url" name="redirect_uri" value="{{.App.PrimaryRedirectURI}}" id="redirect-uri">
+						</div>
+						<button class="ui green button">
+							{{.locale.Tr "settings.save_application"}}
+						</button>
+					</form>
+				</div>
+			</div>
+		</div>
+	</div>
+</div>
+<div class="ui small basic delete modal" id="delete-oauth2-application">
+	<div class="ui icon header">
+		{{svg "octicon-trash"}}
+		{{.locale.Tr "settings.remove_oauth2_application"}}
+	</div>
+	<div class="content">
+		<p>{{.locale.Tr "settings.remove_oauth2_application_desc"}}</p>
+	</div>
+	{{template "base/delete_modal_actions" .}}
+</div>
```

Comment:
What's the code called by?

Reply:
Yup, which code exactly uses this block? Is there any code using the id `delete-oauth2-application`?

Reply:
How could it be? No code is using this id

Reply:
And, when you click the Delete button in the apps list, it opens the id `remove-gitea-oauth2-application`, the dialog in the list page, not in the edit page.